modules/gapi/include/opencv2/gapi/streaming/cap.hpp

```diff
@@ -47,8 +47,17 @@ namespace wip {
 class GCaptureSource: public IStreamSource
 {
 public:
-    explicit GCaptureSource(int id) : cap(id) { prep(); }
-    explicit GCaptureSource(const std::string &path) : cap(path) { prep(); }
+    using Properties = std::unordered_map<int, double>;
+    explicit GCaptureSource(int id, const Properties &properties = {})
```

Comment:
Well, I wish it was separate method `set` but properties must be set before the first frame is read (`prep()`)